### PR TITLE
[IMP] Add pylint check missing newline in extra extension files.

### PR DESCRIPTION
**Pull request by lescobarvx**
[IMP] Add pylint check trailing newlines in other extension files.

### CODE REVIEW

File: pylint_odoo/checkers/modules_odoo.py [hunk 1]
```diff
@@ -59,11 +57,17 @@
...
+        'Trailing newlines in "%s" file',
+        'trailing-newlines-otherfiles',
+        settings.DESC_DFLT
+    ),
 }
 
 
 DFTL_README_TMPL_URL = 'https://github.com/OCA/maintainer-tools' + \
     '/blob/master/template/module/README.rst'
+DFTL_TRAILING_EXT = ['xml', 'csv', 'po', 'js', 'mako']
```
Comment (moylop260) [inline pylint_odoo/checkers/modules_odoo.py]: Could you add this value in option section to have it as configuration file.

File: pylint_odoo/checkers/modules_odoo.py [hunk 2]
```diff
@@ -151,11 +155,27 @@ def _check_duplicate_id_csv(self):
             return False
         return True
 
+    def _check_trailing_newlines_otherfiles(self):
+        """Check trailing newlines in other ext files (.xml, .csv, .po)
+        :return: False if exists errors and
+                 add list of errors in self.msg_args
+        """
+        self.msg_args = []
+        for type_file in DFTL_TRAILING_EXT:
```
Comment (moylop260) [inline pylint_odoo/checkers/modules_odoo.py]: NOTE: If the last comment is applied you will need change this value by option

File: pylint_odoo/checkers/modules_odoo.py [hunk 3]
```diff
@@ -151,11 +155,27 @@ def _check_duplicate_id_csv(self):
...
+            for ext_file in self.filter_files_ext(type_file, relpath=False):
+                last_line = os.stat(ext_file).st_size > 0 and \
+                    file(ext_file, "r").readlines()[-1] or ''
```
Comment (moylop260) [inline pylint_odoo/checkers/modules_odoo.py]: Could you seek to last line directly to avoid even read full file?

Comment (moylop260) [inline pylint_odoo/checkers/modules_odoo.py]: FYI I tested with follow command:
```python
import os
fp = open('PATH_FILE', 'r')
fp.seek(-1, os.SEEK_END)
print fp.read()  # output: ['\n']
```

screen shot 2016-04-18 at 6 31 57 pm

**Comment by moylop260** [comment]
FYI I have created follow tests of performance: https://gist.github.com/moylop260/0dd9350152a6c984285e94db53ea3572

**Comment by lescobarvx** [comment]
@moylop260 
the changes are ready

**Comment by moylop260** [comment]
PR to OCA

Please, add a time for the same test case of performance with the new if of size.
And create other one with a try-except (without if of size) to know what is better

**Comment by lescobarvx** [comment]
Test case with  "if size > 0":
test_case_with_size


Test case with "try / except":
test_with_tryexcept

File: pylint_odoo/test/main.py [hunk 4]
```diff
@@ -28,7 +29,7 @@ def setUp(self):
             self.paths_modules.append(os.path.join(root, path))
         self.default_extra_params = [
             '--disable=all',
-            '--enable=odoolint,pointless-statement',
+            '--enable=odoolint,pointless-statement,trailing-newlines',
```
Comment (moylop260) [inline pylint_odoo/test/main.py]: s/trailing-newlines'/missing-newline-extrafiles/g
New name

**Comment by moylop260** [comment]
Could you rebase it and fix conflicts?

**Pull request merged**